File: L3/S1/TOO/TP4/src/tp4/TirageLoto.java
package tp4;

import java.util.Random;

public class TirageLoto 
{
	public static final int _nombre=48;
	private int[] _resultatTirage;
	
	public TirageLoto()
	{
		_resultatTirage=new int[_nombre];
	}
	
	public void tirage()
	{
		for(int i=0; i<_nombre; i++)
		{
			_resultatTirage[i]=tirageInterne(i);
		}
	}
	
	private int tirageInterne(int i)
	{
		int randomNumber = 0;
		Random rand = new Random(); 
		randomNumber = rand.nextInt(_nombre + 1);
		boolean yetDrawn=false;
		
		for(int j=0; j<i; j++)
		{
			if(this._resultatTirage[j]==randomNumber)
			{
				yetDrawn=true;
				break;
			}
		}
		if(yetDrawn)
		{
			return tirageInterne(i);
		}
		return randomNumber;
		
	}
	
	@Override
	public String toString()
	{
		String tirage=new String();
		for(int value : this._resultatTirage)
		{
			tirage+=value+"\n";
		}
		return tirage;		
	}
}

File: L3/S1/TOO/TP4_CPP/Tirage_loto.cpp
#include <cmath> // 'std::abs', etc.
#include <cstdlib> // 'std::srand', etc.
#include <ctime>

#include "Tirage_loto.h"

const int Tirage_loto::_Nombre = 48;

Tirage_loto::Tirage_loto() {
    _resultat_tirage = new int[_Nombre];
}

Tirage_loto::~Tirage_loto() {
    delete[] _resultat_tirage;
}

void Tirage_loto::tirage() {
    time_t t;
    std::srand(std::time(&t));
    for (int i = 0; i < _Nombre; i++) _resultat_tirage[i] = tirage_interne(i);
}

int Tirage_loto::tirage_interne(const int& i) const {
    int resultat = std::abs(std::rand() % 49); // On tire un nombre positif entre 0 et 49
    bool deja_tire = false; // On fait l'hypothèse que ce nombre n'a pas encore été tiré...
    for (int j = 0; j < i; j++) { // On regarde le début du tableau pour vérifier ce qui a déjà été tiré...
        if (resultat == _resultat_tirage[j]) { // Si déjà tiré...
            deja_tire = true;
            break; // On arrête brutalement la boucle 'for', inutile d'aller plus loin...
        }
    }
    if (deja_tire) return tirage_interne(i); // On recommence le tirage
    return resultat; // Sinon, c'est bon
}